src/net/primitives.c

```diff
@@ -205,21 +203,21 @@ int8_t _z_undeclare_subscriber(_z_subscriber_t *sub) {
 /*------------------ Queryable Declaration ------------------*/
 _z_queryable_t *_z_declare_queryable(_z_session_t *zn, _z_keyexpr_t keyexpr, bool complete,
                                      _z_questionable_handler_t callback, _z_drop_handler_t dropper, void *arg) {
-    _z_questionable_t *rq = (_z_questionable_t *)z_malloc(sizeof(_z_questionable_t));
-    rq->_id = _z_get_entity_id(zn);
-    rq->_key = _z_get_expanded_key_from_key(zn, _Z_RESOURCE_IS_LOCAL, &keyexpr);
-    rq->_complete = complete;
-    rq->_callback = callback;
-    rq->_dropper = dropper;
-    rq->_arg = arg;
-    if (_z_register_questionable(zn, rq) < 0) {
+    _z_questionable_t q;
+    q._id = _z_get_entity_id(zn);
+    q._key = _z_get_expanded_key_from_key(zn, _Z_RESOURCE_IS_LOCAL, &keyexpr);
+    q._complete = complete;
+    q._callback = callback;
+    q._dropper = dropper;
+    q._arg = arg;
+    if (_z_register_questionable(zn, &q) < 0) {
```

Comment:
`z_questionable_t` here can be in the stack avoiding additional heap allocations.
Only when adding it into the list of questionables, it will be copied into the heap by means of the smart pointer creation.